### ISSUE TITLE
Phone Number Formatting

### ISSUE THREAD

**Issue by CWalkden** [tendenci/tendenci#1005]
In the Profiles module the phone number is displayed to a user in the format: (XXX) XXX-XXXX
In Australia the format should be either: XX XXXX XXXX (for landline) or: XXXX XXX XXX (for mobile).  There are other variations as well, but these would be my preference.
There are some localisation settings in Tendenci, is it possible to apply this to this field?

**Comment by jennyq** [comment]
Yes, it's possible. I'm not sure which setting can be used. Maybe just use a new setting? Then the [phonenumber](https://github.com/tendenci/tendenci/blob/master/tendenci/apps/base/templatetags/base_filters.py#L458) filter can be updated based on the setting. You're welcome to submit a pull request.

**Comment by rob-hills** [comment]
@jennyq , would you be open to modifying the [phonenumber](https://github.com/tendenci/tendenci/blob/master/tendenci/apps/base/templatetags/base_filters.py#L458) filter to use the [PyPi Phonenumbers](https://pypi.org/project/phonenumbers/) Library?  I would be happy to have a go at doing a pull request for this.

**Comment by jennyq** [comment]
The python phonenumbers looks like a good library. My only concern is the memory overhead. Maybe just use [phonenumberslite](https://pypi.org/project/phonenumberslite/) version as they suggested. @eschipul, what do you think?

**Comment by eschipul** [comment]
@jennyq I agree that as @CWalkden pointed out, given most of our client base is international, properly displaying the phone number is important. If the phonenumberslite module works to keep it fast, then that seems reasonable.

It might also be that if we are storing the phone number in a char field that we would only need to load either library on add/edit screens - and not displays, views or reporting.

**Comment by jennyq** [comment]
Thanks @eschipul ! @rob-hills You're welcome to proceed with [phonenumberslite](https://pypi.org/project/phonenumberslite/).

**Comment by rob-hills** [comment]
@jennyq I've created a phonenumberslite branch in our AEVA fork of the Tendenci code and done some basic testing.  I had originally put a couple of new configuration settings into the settings.py file but our other website admins wanted them in apps/site_settings/settings.json file instead so it could be modified from the Web Admin interface.
I've put my new settings into the AEVA fork apps/site_settings/settings.json file and installed onto our test server from my phonenumberslite branch the way I do normally when testing new code.  However the new settings aren't present on the settings/site/global webpage of our site.  Is there a special trick to deploy changes to the global site_settings settings.json file on a test website?

**Comment by rob-hills** [comment]
I've done a bit more sleuthing and established that in our test site there are two copies of this file:
... environment/lib/python3.8/site-packages/tendenci/apps/site_settings/settings.json
... environment/src/tendenci/tendenci/apps/site_settings/settings.json

The settings.json in src contains my updates, the one in site-packages doesn't so I'm guessing that is the one in use by our tendenci site.

**Comment by jennyq** [comment]
> The settings.json in src contains my updates, the one in site-packages doesn't so I'm guessing that is the one in use by our tendenci site.

Hmm, interesting. I'm not sure why you have two copies of that file. Anyway, the file "apps/site_settings/settings.json" is the correct place to add your phone number setting.

**Comment by rob-hills** [comment]
> > The settings.json in src contains my updates, the one in site-packages doesn't so I'm guessing that is the one in use by our tendenci site.
> 
> Hmm, interesting. I'm not sure why you have two copies of that file. Anyway, the file "apps/site_settings/settings.json" is the correct place to add your phone number setting.

@jennyq should this file (apps/site_settings/settings.json) contain our current site settings?

As far as I can tell, it doesn't.  For example, on our Test Server "Global Settings" page (https://test.aeva.asn.au/settings/site/global/) we have "Australia" set as the country for our site.  However when I check the apps/site_settings/settings.json file, it has the following section for "Country":
```
    {
        "default_value": "US",
        "description": "Enter the Country for your site.",
        "data_type": "string",
        "update_dt": "2010-08-12 17:19:13",
        "input_type": "select",
        "client_editable": true,
        "name": "defaultcountry",
        "value": "US",
        "label": "Country",
        "input_value": "<country_list>",
        "parent_id": 0,
        "scope": "site",
        "scope_category": "global",
        "store": true,
        "updated_by": ""
    },

```
This looks suspiciously like the default values that come at installation time.

I'm guessing changes to these settings are stored somewhere else than the apps/site_settings/settings.json file?

**Comment by rob-hills** [comment]
I tried adding my two new settings to the settings.json file in our source directory:
`environment/src/tendenci/tendenci/apps/site_settings/settings.json`
and then running:
```
python manage.py update_settings site_settings
```
However the Site Settings web page is unchanged and the new settings haven't appeared in the DB:
```
select * from site_settings_setting where lower(name) like '%phone%';
```

So, next I tried adding my two new settings to the settings.json file in our site packages directory:
`environment/lib/python3.8/site-packages/tendenci/apps/site_settings/settings.json`
and again running the update program.

Unfortunately this made no difference, neither to the Site Settings web page, nor to the DB records.

**Comment by rob-hills** [comment]
@jennyq, as I [mentioned above](https://github.com/tendenci/tendenci/issues/1005#issuecomment-2541584273), we have two of these files so I'm not sure which one is in use by our system.  It would be helpful if debug logging included the full path of the settings file being loaded.

Anyway, **both** of our settings files have the following two entries at the end:

```
    },
    {
        "default_value": "US",
        "description": "Enter the assumed phone number region (ISO 3116-1) where an international code isn't specified",
        "data_type": "string",
        "update_dt": "2024-12-12 23:00:00",
        "input_type": "text",
        "client_editable": true,
        "name": "phone_number_region",
        "value": "US",
        "label": "Phone Number Region",
        "input_value": "string",
        "parent_id": 0,
        "scope": "site",
        "scope_category": "global",
        "store": true,
        "updated_by": ""
    },
    {
        "default_value": "(###) ###-####",
        "description": " All '#' characters are replaced with a phone number digit, all other characters are treated as literals",
        "data_type": "string",
        "update_dt": "2024-12-12 23:00:00",
        "input_type": "text",
        "client_editable": true,
        "name": "phone_number_pattern",
        "value": "(###) ###-####",
        "label": "Phone Number Pattern",
        "input_value": "string",
        "parent_id": 0,
        "scope": "site",
        "scope_category": "global",
        "store": true,
        "updated_by": ""
    },
]
```

Note, I have included the closing "curly-brace" character from what was previously the last setting in the file.  I have also included the closing square-bracket for the whole settings file.

I note that the final closing curly-brace is followed by a comma, which I am used to doing for Python lists etc.  Does json support this?  Could this trailing comma be messing things up?

**Comment by jennyq** [comment]
RE "the final closing curly-brace is followed by a comma", yes, the trailing comma needs to be removed.

**Comment by rob-hills** [comment]
Pull Request #1340 submitted to resolve this issue.
